packages/ckeditor5-html-support/src/integrations/table.js

```diff
@@ -0,0 +1,164 @@
+		const dataFilter = editor.plugins.get( DataFilter );
+
+		dataFilter.on( 'register:table', ( evt, definition ) => {
+			if ( definition.model !== 'table' ) {
+				return;
+			}
+
+			schema.extend( 'table', {
+				allowAttributes: [
+					'htmlAttributes',
+					// Figure, thead and tbody elements don't have model counterparts.
+					// We will be preserving attributes on table element using these attribute keys.
+					'htmlFigureAttributes', 'htmlTheadAttributes', 'htmlTbodyAttributes'
+				]
+			} );
+
+			conversion.for( 'upcast' ).add( disallowedAttributesConverter( definition, dataFilter ) );
+			conversion.for( 'upcast' ).add( viewToModelTableAttributeConverter( dataFilter ) );
+			conversion.for( 'downcast' ).add( modelToViewTableAttributeConverter() );
+
+			evt.stop();
+		} );
+
+		dataFilter.on( 'register:figure', () => {
+			conversion.for( 'upcast' ).add( consumeTableFigureConverter() );
+		} );
+	}
+}
+
+// View-to-model conversion helper preserving allowed attributes on {@link module:table/table~Table Table}
+// feature model element.
+//
+// @private
+// @param {module:html-support/datafilter~DataFilter} dataFilter
+// @returns {Function} Returns a conversion callback.
+function viewToModelTableAttributeConverter( dataFilter ) {
+	return dispatcher => {
+		dispatcher.on( 'element:table', ( evt, data, conversionApi ) => {
+			const viewTableElement = data.viewItem;
+
+			preserveElementAttributes( viewTableElement, 'htmlAttributes' );
+
+			const viewFigureElement = viewTableElement.parent;
+			if ( viewFigureElement.is( 'element', 'figure' ) ) {
+				preserveElementAttributes( viewFigureElement, 'htmlFigureAttributes' );
+			}
+
+			for ( const childNode of viewTableElement.getChildren() ) {
+				if ( childNode.is( 'element', 'thead' ) ) {
+					preserveElementAttributes( childNode, 'htmlTheadAttributes' );
+				}
+
+				if ( childNode.is( 'element', 'tbody' ) ) {
+					preserveElementAttributes( childNode, 'htmlTbodyAttributes' );
+				}
+			}
+
+			function preserveElementAttributes( viewElement, attributeName ) {
+				const viewAttributes = dataFilter._consumeAllowedAttributes( viewElement, conversionApi );
+
+				if ( viewAttributes ) {
+					conversionApi.writer.setAttribute( attributeName, viewAttributes, data.modelRange );
+				}
+			}
+		}, { priority: 'low' } );
+	};
+}
+
+// Model-to-view conversion helper applying attributes from {@link module:table/table~Table Table}
+// feature.
+//
+// @private
+// @returns {Function} Returns a conversion callback.
+function modelToViewTableAttributeConverter() {
+	return dispatcher => {
+		addAttributeConversionDispatcherHandler( 'table', 'htmlAttributes' );
+		addAttributeConversionDispatcherHandler( 'figure', 'htmlFigureAttributes' );
+		addAttributeConversionDispatcherHandler( 'thead', 'htmlTheadAttributes' );
+		addAttributeConversionDispatcherHandler( 'tbody', 'htmlTbodyAttributes' );
+
+		function addAttributeConversionDispatcherHandler( elementName, attributeName ) {
```

Comment:
Same thing as in line 92.

Reply:
https://github.com/ckeditor/ckeditor5/pull/10104#discussion_r669675048